routing/missioncontrol.go

```diff
@@ -383,6 +392,31 @@ func (m *MissionControl) ReportPaymentFail(paymentID uint64, rt *route.Route,
 		route:            rt,
 	}
 
+	return m.processPaymentResult(result)
+}
+
+// ReportPaymentSuccess reports a successful payment to mission control as input
+// for future probability estimates.
+func (m *MissionControl) ReportPaymentSuccess(paymentID uint64,
+	rt *route.Route) error {
+
+	timestamp := m.now()
+
+	result := &paymentResult{
+		timeFwd:   timestamp,
+		timeReply: timestamp,
+		id:        paymentID,
+		success:   true,
```

Comment:
Could we instead use the existing failure fields of `paymentResult` to determine if the payment succeeded?

Reply:
I don't think there is an existing field that can serve this purpose. `failureSourceIdx` can be nil for failure with an unknown source and `failure` can be nil for unknown failure messages.